src/domain/Tables/Table/Table.tsx

```diff
@@ -1,55 +1,45 @@
-import { mapValues } from 'lodash'
-import React, { useEffect, useState } from 'react'
-import { Transition, TransitionGroup } from 'react-transition-group'
+import { isEqual } from 'lodash'
+import React, { useCallback, useEffect, useMemo, useState } from 'react'
 
 import { Props } from '../../../common'
 import {
-  FontAwesomeIconButton,
   IFontAwesomeIconButtonProps
 } from '../../Buttons/FontAwesomeIconButton/FontAwesomeIconButton'
 import { TableRowVariant } from './services/colors'
 import {
   getSelectAllTableCheckboxInputValue,
   getUpdatedAllSelectableRows,
-  handleSelectionChanged,
-  validateProps
+  handleSelectionChanged
 } from './services/helper'
 import {
   StyledEmptyStateCell,
   StyledEmptyStateRow,
-  StyledFontAwesomeIconButtonWrapper,
   StyledTHead,
-  StyledTable, StyledTableWrapper
+  StyledTable,
+  StyledTableWrapper
 } from './services/style'
+import {
+  ColumnAlignment,
+  ColumnSize,
+  ColumnSortDirection,
+  IColumnProps,
+  IColumnSorts,
+  IRowProps,
+  ISelectedRows,
+  InteractionType
+} from './services/types'
```

Comment:
All types were moved to `types.ts` to fix the circular import of table (likely to result in issues as written, plus vscode complains heavily about it)